### ISSUE TITLE
False positive with type variable seemingly not being bound correctly when return type is inferred

### ISSUE THREAD

**Issue by matangover** [microsoft/pylance-release#2785]
## Environment data

-   Language Server version: 2022.4.3 (pyright 88f2a424)
-   OS and version: macOS 10.15.7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.0

## Expected behavior

Pyright shouldn't give an error when calling `mylen(mylist)`

## Actual behavior

Pyright gives error as seen in the following screenshot:

<img width="900" alt="image" src="https://user-images.githubusercontent.com/5028956/167360136-fb4b001f-c700-484c-b486-81736da4ed93.png">

If line 11 in `ptest1.py` is uncommented, the bug disappears.

The bug also disappears if `def __getitem__(self, idx):` is changed to `def __getitem__(self, idx) -> T:`.

## Code Snippet / Additional information

File `ptest1.py`:

```python
from _typeshed import SupportsLenAndGetItem
from ptest2 import MyList

def mylen(x: SupportsLenAndGetItem[str]):
    return len(x)


mylist = MyList(["a", "b", "c"])
reveal_type(mylist)
# Bug disappears if following line is uncommented:
# reveal_type(mylist[0])
mylen(mylist)
```

File `ptest2.py`:

```python
from typing import Generic, TypeVar


T = TypeVar("T")

class MyList(Generic[T]):
    _list: list[T]

    def __init__(self, l: list[T]):
        self._list = l

    def __getitem__(self, idx):
        return self._list[idx]

    def __len__(self):
        return len(self._list)
```

**Comment by erictraut** [comment]
Thanks for reporting the problem.

This bug was already fixed last week in pyright, and the fix will be in the next release of pylance.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.5.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202251-11-May-2022)
